app/controllers/warehouse_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class WarehouseController < ApplicationController
+  respond_to :json
+
+  def index
+    warehouses = Warehouse.where(company_id: params[:company_id])
+    json = []
+    warehouses.each do |warehouse|
+      json << {
+        warehouse: warehouse,
+        user: warehouse.users.find_by(user_role_id: 4)
```

Comment:
I think it needs to be replaced with "where" so that the select returns an array
it is also desirable to get rid of 'Magic number'